org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/ScalaDebugCache.scala

```diff
@@ -244,15 +265,37 @@ protected[debug] class ScalaDebugCacheActor(debugCache: ScalaDebugCache, debugTa
       classLoaded(e)
       reply(false)
     case LoadedNestedTypes(outerTypeName) =>
-      reply(LoadedNestedTypesAnswer(getLoadedNestedTypes(outerTypeName)))
+      val replyTo = sender
+      become {
+        case LoadedNestedTypesUpdate(outerTypeName, cache) =>
+          updateCacheWith((outerTypeName, cache))
+          replyTo ! LoadedNestedTypesAnswer(cache.types)
+          unbecome()
+      }
+      findCache(outerTypeName).onSuccess {
+        case (typeName, cache) =>
+          ScalaDebugCacheActor.this ! LoadedNestedTypesUpdate(typeName, cache)
+      }
     case AddClassPrepareEventListener(actor, outerTypeName) =>
-      addClassPreparedEventListener(actor, outerTypeName)
-      reply(true)
+      val replyTo = sender
+      become {
```

Comment:
I don't fully understand the code changes here. Why did you need to create a new message and send it back to self in this way?